llmware/parsers.py

```diff
@@ -3066,6 +3066,151 @@ def basic_library_duplicate_check(self, fn):
 
         return in_library
 
+    def parse_delimiter_config(self,fp, fn, cols=None, mapping_dict=None, delimiter=","):
+
+        """ Designed for intake of a 'pseudo-db csv table' and will add rows to library with mapped keys.
+
+        special feature here is any delimiter is possible there is no restriction in lenght
+            so you could do a delimiter like ##superspecialdelimiter##
+
+        Inputs:
+            -- csv folder path + csv file name
+            -- cols = # of expected column entries in each row of the CSV
+            -- mapping dict = assigns key names to columns, starting with 0 for first column
+                e.g., {"text": 4, "doc_ID": 2, "key1": 3}
+
+        Requirements:
+            -- must have a "text" key in the mapping dictionary
+            -- optional doc_ID and block_ID - if found, will over-write the normal library indexes
+            -- all other keys will be saved as 'metadata' and added to the library block row in "special_field1"
+
+        Note: this feature is currently only supported for Mongo - SQL DB support will follow.
+        """
+
+        # method requires library loaded in the Parser
+
+        if not self.library:
+            raise LLMWareException(message="Parsing of a configured CSV file requires a library object to "
+                                           "be connected to the parser state.")
+
+        #   if found in mapping dict, then will over-write
+        reserved_keys = ["text", "doc_ID", "block_ID"]
+
+        rejected_rows = []
+
+        if not mapping_dict:
+            raise LLMWareException(message="Parsing of a configured CSV file requires a mapping dictionary so that "
+                                           "the table attributes can be properly mapped.")
+
+        if not cols:
+            raise LLMWareException(message="Parsing of a configured CSV file requires a defined column structure and "
+                                           "a specified number of columns to ensure accurate mapping.")
+
+        # file type
+        ft = fn.split(".")[-1].lower()
+        if ft == "tsv":
+            delimiter="\t"
+
+        # will iterate through csv file
+        input_file = os.path.join(fp, fn)
+
+        f = open(input_file, 'r', encoding='utf-8-sig',errors='ignore')
+        filedata = f.readlines()
+
```

Comment:
Please change this to
```suggestion
        with open(input_file, 'r', encoding='utf-8-sig', errors='ignore') as f:
            filedata = f.readlines()

```

Reply:
there is no technical advance except for different syntax